src/drivers/imu/adis16497/ADIS16497.cpp

```diff
@@ -247,7 +249,19 @@ ADIS16497::probe()
 			PX4_DEBUG("PRODUCT: %X", product_id);
 
 			if (self_test()) {
-				return PX4_OK;
+
+				// Switch to config page
+				write_reg16(PAGE_ID, 0x03);
```

Comment:
You don't need to switch back?

Reply:
Wherever config params are touched, the code always switches to relevant page first. So don't need to explicitly do it here.